common/cache/lru.go

```diff
@@ -131,14 +136,28 @@ func New(maxSize int, opts *Options) Cache {
 		opts = &Options{}
 	}
 
-	return &lru{
+	cache := &lru{
```

Comment:
To me, size and count is either/or, meaning that if it's size based, we don't care about the count, as long as it stays within size limit.

Reply:
Agree, if maxSize is provided then it is size based only and count based otherwise. And if size based, lets just check for an upper limit for count to prevent the infinite growing, for safety sake.